snapcraft/storeapi/assertions.py

```diff
@@ -0,0 +1,262 @@
+        """
+        new_assertion = deepcopy(self.assertion)
+        new_assertion['developers'] = developers
+
+        # The revision should be incremented, to avoid `invalid-revision`
+        # errors.
+        new_assertion['revision'] = str(int(
+            self.assertion.get('revision', '0'))+1)
+
+        # There is a possibility that the `authority-id` to be `canonical`,
+        # which should be changed to the `publisher_id` to match the signing
+        # key.
+        new_assertion['authority-id'] = self.publisher_id
+
+        new_instance = DeveloperAssertion(snap_name=self._snap_name,
+                                          signing_key=self._signing_key)
+        # Reference the already fetched information
+        new_instance._account_info = self._account_info
+        new_instance.assertion = new_assertion
+        return new_instance
+
+    def get(self):
+        """Returns a dict containing the developer assertion for snap_name.
+
+        The data that comes from the store query looks as follows:
+        {'snap_developer': {
+             'type': 'snap-developer',
+             'authority-id': <account_id of the publisher>,
+             'publisher-id': <account_id of the publisher>,
+             'snap-id': 'snap_id',
+             'developers': [{
+                 'developer-id': 'account_id of dev-1',
+                 'since': '2017-02-10T08:35:00.390258Z'
+                },{
+                 'developer-id': 'account_id of dev-2',
+                 'since': '2017-02-10T08:35:00.390258Z',
+                 'until': '2018-02-10T08:35:00.390258Z'
+                }],
+             }
+        }
+
+        The assertion is saved without the snap_developer header.
+
+        :returns: the latest developer assertion corresponding to snap_name.
+        :rtype: dict
+        """
+        try:
+            self.assertion = super().get()
+        except errors.StoreValidationError as e:
+            if e.error_list[0]['code'] != 'snap-developer-not-found':
+                raise
+            self.assertion = {
+                'type': 'snap-developer',
+                'authority-id': self.publisher_id,
+                'publisher-id': self.publisher_id,
+                'snap-id': self.snap_id,
+                'developers': [],
+            }
+        # A safeguard to operate easily on the assertion
+        if 'developers' not in self.assertion:
+            self.assertion['developers'] = []
+        return self.assertion
+
+    def get_developers(self):
+        """Return a copy of the current developers listed in the assertion.
+
+        :returns: a list of developers.
+        :rtype: list
+        """
+        return self.get()['developers'].copy()
+
+    def is_equal(self, other_assertion):
+        """Determines equality of developer lists in each assertion.
+
+        Returns True if the list of developers in this instances assertion list
+        is equal to the list from other_assertion.
```

Comment:
I would put this in :returns:, but just a personal preference, not requesting a change.